python/aitemplate/testing/detect_target.py

```diff
@@ -32,22 +32,21 @@
 
 def _detect_cuda():
     try:
-        proc = Popen(
-            ["nvidia-smi", "--query-gpu=gpu_name", "--format=csv"],
-            stdout=PIPE,
-            stderr=PIPE,
-        )
-        stdout, stderr = proc.communicate()
-        stdout = stdout.decode("utf-8")
-        if "H100" in stdout:
+        import pycuda.driver as drv
+
+        drv.init()
+        major, minor = drv.Device(0).compute_capability()
```

Comment:
should we use env CUDA_VISIBLE_DEVICES for the device number?

Reply:
`drv.Device(0)` points to the first visible device. See detailed comment below.